js/utility/mDDuration.js

```diff
@@ -0,0 +1,6 @@
+export default function calculateDuration(startdate) {
+  const enddate = new Date().getTime();
```

Comment:
Let's put a navigation guard at the top of the function to check if `startDate` is smaller than `endDate`. 
```suggestion
export default function calculateDuration(startDate) {
  const startDate = Date.now();
  
  if(startDate > endDate){
    return -1
  }
```

Now let's use the returned value to determine whether the `calculateDuration` function is working as expected or not.

Reply:
I have a confusion regarding this. the startDate will be generated and stored at the time of task creation and the make-task-state-to-done button can only be visible after that. And the endDate can only be generated when make-task-state-to-done button is clicked. Since it is a SynchronousI process I think startDate will always be smaller

Reply:
It's really easy to assume that `endDate` should always be bigger than `startDate`, but what is hard to assume is human behaviour in general. As the `calculateDuration` function is a generalised helper function, in any project if we don't put guards, devs can pass future date as `startDate` and receive unexpected behaviour.

Reply:
Insightful